### PR TITLE
Task/fix and update samples testsuite

### PR DESCRIPTION
**Pull request by drl-max**
I wanted to investigate an issue I was having with tab-delimited data in my application and figured that I could leverage the module's test suite to, at the very least, prove that it was my application that was the problem. However, when I tried to leverage the `~/samples/tsv.js` file I started doubting if it was even being executed. 

This lead me to the `samples.coffee` script and the fact that the `filter` clause didn't seem correct to me, which I proved to be the case. Once that was fixed, the files started getting executed. I addressed all of the failing tests, tweaked some of them, and addressed a deprecation warning. Each commit on this branch describes the change.

### CODE REVIEW

File: samples/tsv.js [hunk 1]
```diff
@@ -1,11 +1,22 @@
...
+  assert.deepStrictEqual(data, [ [ '1', '2', '3' ], [ 'a', 'b', 'c' ] ]);
+});
 
-parse( "1	2	3\ra	b	c", {delimiter: '\t'}, function(err, data){
+parse(
+`"1"	"2"	"3"
+"a"	"b"	"c"`, {
+  delimiter: '\t',
+  quote: "\"",
+  columns: true
```
Comment (wdavidw) [inline samples/tsv.js:18]: I am ok and grateful of the refactoring but I wish to preserve the original options, without `quote` and `columns`.

File: test/samples.coffee [hunk 2]
```diff
@@ -5,15 +5,12 @@ path = require 'path'
 {exec} = require 'child_process'
 each = require 'each'
   
-it 'samples', ->
+it 'samples', -> 
   dir = path.resolve __dirname, '../samples'
   samples = await fs.readdir dir
-  each samples.filter( (sample) -> /\.js/.test.sample)
```
Comment (wdavidw) [inline test/samples.coffee:11]: Right, this definitely doesn't look right!

**Review by wdavidw — CHANGES_REQUESTED** [review]
One minor change, thank you very much for taking the time

File: samples/tsv.js (hunk 1 above)
Comment (drl-max) [reply samples/tsv.js]: Okay, I went ahead and removed that test case and reverted my change to the formatting of the original test case.  For what it's worth, your original test case was still there, just above this particular one.  The original one didn't cover the `columns` use case, so I thought it was a variant worth contributing, but that's fine. I _think_ I've discovered a bug specifically pertaining to tab-delimited values, but I haven't shored up my bug report yet so I'll hold off bringing that up until later.

**Comment by wdavidw** [comment]
Thank you for this great contribution. I have update the website to reflect the changes.

**Pull request merged**